src/lib/core/ReferenceCounted.h

```diff
@@ -41,7 +41,7 @@ class DeleteDeletor
  * A reference counted object maintains a count of usages and when the usage
  * count drops to 0, it deletes itself.
  */
-template <class SUBCLASS, class DELETOR = DeleteDeletor<SUBCLASS>>
+template <class SUBCLASS, class DELETOR = DeleteDeletor<SUBCLASS>, int INITIAL = 1>
```

Comment:
Under what circumstances would initial not be 1?

I see ExchangeContext starts with 0, however could you explain why? I assume it does a Retain() after allocation or within the construction, however could we instead remove that extra retain and keep initial 1?

My reasoning would be that I expect any new(Foo) to be referenced counted to exaclty 1 post allocation.

If we decide to keep this, nitpicks: type should be uint32_t, naming should be kInitial.

Reply:
When we initializing the pool, it will be 0.

When constructed inside a pool, all objects are not considered as being used but free and ready for allocation.

Reply:
This explanation would be useful in the doccomment.

Please name it using a `kCamelCase` naming style (it is a constant defined as a typesafe data rather than a `#define`)

Reply:
Done